data/lib/service/tournament/tournament_service.dart

```diff
@@ -31,4 +38,31 @@ class TournamentService {
       throw AppError.fromError(error, stack);
     }
   }
+
+  Stream<List<TournamentModel>> streamCurrentUserRelatedMatches() {
+    if (_currentUserId == null) {
+      return Stream.value([]);
+    }
+    final currantMember = TournamentMember(id: _currentUserId ?? 'INVALID ID');
+
+    final memberContains = [
+      currantMember.copyWith(role: TournamentMemberRole.organizer).toJson(),
+      currantMember.copyWith(role: TournamentMemberRole.admin).toJson(),
+    ];
+    final filter = Filter.or(
+      Filter(FireStoreConst.createdBy, isEqualTo: _currentUserId),
+      Filter(FireStoreConst.members, arrayContainsAny: memberContains),
+    );
+
+    return _tournamentCollection
+        .where(filter)
+        .snapshots()
+        .asyncMap((snapshot) async {
+      return await Future.wait(
+        snapshot.docs.map((mainDoc) async {
+          return mainDoc.data();
+        }).toList(),
+      );
```

Comment:
there isn't any async operation here, then why do we need asyncMap? can't we use map only?